tasks/cycles/cycles.js

```diff
@@ -13,24 +13,53 @@
 function calculateDiscount(redemption) {
   let discount;
 
-  //PLACE YOUR CODE HERE
-
-  return discount;
+if (redemption>=0, redemption <=350){
+  discount = ('0%');
+} else if (redemption>=351, redemption <=1350){
+  discount = ('15%');
+} else if (redemption>=1351, redemption <=2700){
+ discount = ('30%');
+} else if (redemption>=2701, redemption <=6500){
+  discount = ('45%');
 }
+          
+      return discount;}
+console.log(calculateDiscount(1000));
+
 
 /** TODO
  * implement factorial algorithm using for, while, do..while operators
  */
-{
-  const i = 10; //10! = 3628800
-}
+
+  const i = 10; 
+  let x=1;
+  let y=1;
+  while (y<i){
+    x=x*y;  y++ ;
+}  console.log(x);
+
+let x=1; 
+for (let y=1; y< 10;  y++)
+{ x=x*y;}
+console.log(x);
+
+const i = 10; 
+let x=1;
+let y=1;
+do  {x=x*y;  y++} 
+while (y<i);
+ console.log(x);
 
 /**
  * return concatenated string from an array of substring
  */
-{
+
   const substr = ["I", " love", " JS"];
-}
+  let text = '';
+  for (let x in substr) {
```

Comment:
it is better to use for...of cycle in case of arrays:
 for (let x of substr) {...}